core/trino-main/src/main/java/io/trino/operator/join/SortedPositionLinks.java

```diff
@@ -79,8 +85,7 @@ public int link(int from, int to)
             // make sure that from value is the smaller one
             if (comparator.compare(from, to) > 0) {
                 // _from_ is larger so, just add to current chain _to_
-                List<Integer> links = positionLinks.computeIfAbsent(to, key -> new IntArrayList());
-                links.add(from);
+                positionLinks.computeIfAbsent(to, NEW_INT_LIST).add(from);
```

Comment:
If it is a hot code maybe just do a manual get + if. Like in the old days

Reply:
In this case I think that would actually be a bit slower, since a get() / put() pair would have to do another hash / probe sequence compared to a single computeIfAbsent call.